src/classes/sequelize-service.ts

```diff
@@ -97,6 +101,36 @@ export class SequelizeService<W extends {}, R extends W, C extends {} = {}> exte
     }, this.errorFactory);
   }
 
+  public async upsert<KC extends keyof C = keyof {}>(object: W, options: TUpsertOptions<R, C> = {}): Promise<R & Pick<C, KC>> {
+    return await SequelizeService.try(async () => {
+      return await this.transaction(options, async () => {
+        const session = new UpsertSession<W, R, C, KC>([object], options, this);
+
+        await this.executeHook(Hook.WILL_UPSERT, session, this._beforeUpsert.bind(this));
+
+        const created = await this.model.upsert(<R>object, options);
+
+        session.setCreated(created);
+
+        let filters: TFilters<R>;
```

Comment:
If set, we only the attributes defined in `fields` to find the row after performing the upsert (because sequelize doesn't return us the row, this is a little unsafe 🙈 )

Not 100% sure if that makes sense, so any better ideas are welcome :)